dowhy/causal_estimators/causalml.py

```diff
@@ -0,0 +1,108 @@
+            self._instrumental_variables = pd.get_dummies(self._instrumental_variables, drop_first=True)
+        else:
+            self._instrumental_variables = []
+
+        #Check if effect modifiers are used
+        self.logger.debug("Effect Modifiers used:" + 
+                        ",".join(self._effect_modifier_names))
+        
+        
+        # Get the class corresponding the the estimator to be used
+        estimator_class = self._get_causalml_class_object(self._causalml_methodname)
+        # Initialize the object
+        # pdb.set_trace()
+        self.estimator = estimator_class(**self.method_params["init_params"])
+        self.logger.info("INFO: Using CausalML Estimator")
+        self.symbolic_estimator = self.construct_symbolic_estimator(self._target_estimand)
+        self.logger.info(self.symbolic_estimator)
+        
+    def _get_causalml_class_object(self, module_method_name, *args, **kwargs):
+        
+        try:
+            (module_name, _, class_name) = module_method_name.rpartition(".")
+            estimator_module = import_module(module_name)
+            estimator_class = getattr(estimator_module, class_name)
+
+        except (AttributeError, AssertionError, ImportError):
+            raise ImportError('Error loading {}.{}. Double-check the method name and ensure that all econml dependencies are installed.'.format(module_name, class_name))
+        return estimator_class
+
+    def _estimate_effect(self):
+        X_names = self._instrumental_variable_names + \
+                self._observed_common_causes_names + \
+                self._effect_modifier_names
+
+        y_name = self._outcome_name # We need to make sure that both are pd.Series, so we take the first element
+        treatment_name = self._treatment_name[0] # As we have only one treatment variable
+        
+        func_args={
+            'X':self._data[X_names],
+            'y':self._data[y_name],
+            'treatment':self._data[treatment_name]
+        }
+
+        arg_names = inspect.getfullargspec(self.estimator.estimate_ate)[0]
+        # pdb.set_trace()
+        matched_args = {
+            arg: func_args[arg] for arg in func_args.keys() if arg in arg_names 
+        }
+        value_tuple = self.estimator.estimate_ate(**matched_args) 
+
+        estimate = CausalEstimate(estimate=value_tuple[0],
+                                  target_estimand=self._target_estimand,
+                                  realized_estimand_expr=self.symbolic_estimator,
+                                  effect_intervals=(value_tuple[1],value_tuple[2]),
+                                  _estimator_object=self.estimator)
+
+        return estimate
+        
+
+    def construct_symbolic_estimator(self, estimand):
+        expr = "b: " + ",".join(estimand.outcome_variable) + "~"
+        # TODO we are conditioning on a postive treatment
+        var_list = estimand.treatment_variable + estimand.backdoor_variables

```

Comment:
A simple addition here is to add (+ effect_modifiers). That makes it consistent with the code above.
That said, we'll need to do a lot more work to derive symbolic estimators separately for each method, which we can leave for later. For example, the current symbolic string is correct only for T-learner, but not for S-learner. For S-learner, we need to construct two such strings (both without t) and then subtract them. 
Anyways, this will become complicated (and does not affect the estimation). So we can maybe create a todo under the estimation project to fix for later.

Reply:
Makes sense. I'll add a TODO for the same